src/crypto/mod.rs

```diff
@@ -0,0 +1,291 @@
+mod error;
+
+use byteorder::{BigEndian, ByteOrder};
+
+use pairing::{CurveAffine, CurveProjective, Engine, Field, PrimeField};
+use rand::{ChaChaRng, Rand, Rng, SeedableRng};
+use ring::digest;
+
+use self::error::{ErrorKind, Result};
+
+/// Returns a hash of the given message in `G2`.
+pub fn hash_g2<E, M>(msg: M) -> E::G2
+where
+    E: Engine,
+    <E as Engine>::G2: Rand,
+    M: AsRef<[u8]>,
+{
+    let digest = digest::digest(&digest::SHA256, msg.as_ref());
+    // The `pairing` crate's `G2` implements `Rand`. We initialize a seedable RNG with the SHA512
```

Comment:
Now the reference to SHA512 is obsolete in the comment.

Reply:
Thanks! I fixed it.